### DISCUSSION TITLE
how iterate through the binaryninja open tabs?

### DISCUSSION THREAD

**Discussion by hanssl** [Vector35/binaryninja-api#2929] [API] [answered]
Prior to the 3.0 release of binja, it was possible to iterate over the open tabs and then switch to the desired one (see the attached screenshot).  This code no longer works with the 3.0 release.  Is there an improved way to perform this operation in 3.0
![Screenshot from 2022-02-02 08-59-48](https://user-images.githubusercontent.com/649957/152398706-19417b0b-2c1c-4e3d-813a-be963d422cf7.png)
?

**Comment by psifertex** [comment, marked as answer]
Yes, this is a result of the UI refactor that changed how the window hierarchy works. Something like this worked for me:

```python
for tab in UIContext.activeContext().getTabs():
  if tab.getTabName() == "ls":
    UIContext.activeContext().activateTab(tab)
```

**Reply by hanssl** [reply]
@psifertex If I run the following in the binaryninja python console then I get back a context:

from binaryninjaui import UIContext
UIContext.activeContext()

However, I run the same code from a python plugin then activeContext() returns 'None'.

**Reply by psifertex** [reply]
Yes, for a UI plugin you'll want to register a UIAction which will get passed a context. Similar to how snippets does it here:

https://github.com/Vector35/snippets/blob/master/__init__.py#L718

**Reply by hanssl** [reply]
thank you
